buildSrc/src/main/java/org/elasticsearch/gradle/DistributionDownloadPlugin.java

```diff
@@ -226,11 +225,20 @@ private static Dependency projectDependency(Project project, String projectPath,
 
     private static String distributionProjectPath(ElasticsearchDistribution distribution) {
         String projectPath = ":distribution";
-        if (distribution.getType() == Type.INTEG_TEST_ZIP) {
-            projectPath += ":archives:integ-test-zip";
-        } else {
-            projectPath += distribution.getType() == Type.ARCHIVE ? ":archives:" : ":packages:";
-            projectPath += distributionProjectName(distribution);
+        switch (distribution.getType()) {
+            case INTEG_TEST_ZIP:
+                projectPath += ":archives:integ-test-zip";
+                break;
+
+            case DOCKER:
```

Comment:
@pugnascotia the download distribution plugin is meant to be used outside of our build,
@rjernst is working on adding support to download some of the artifacts that are not already downloadable.  I think in the case of docker, this could be set up differently to make that work easier.
For example, `ensureImageIsLoaded` from the test should be part of this plugin, since docker is distributed trough the registry, so there will be nothing to do here for the released versions. 

Alternatively @rjernst,  it might be more straight forward and easier to maintain to split the download plugin into two. One that deals exclusively with versions that are an actual download, and restrict that to some distribution types and one that deals with unreleased versions, with the lather being applied only as part of our build. 

I don't necessarily see this as reason  not to merge this PR.

Reply:
Yes, I feel that we should merge this and come back to some of these issues. At least we'll have something build upon.